### PR TITLE
Improve the usage experience with IsInRangeConverter in XAML

### PR DESCRIPTION
**Pull request by GeorgeLeithead**
### Description of Change ###

When initially written, the `IsInRangeConverter` used `CompareConverter` as a guide/basis.  Since then, the `CompareConverter` has been changed to improve the usage experience in XAML.  See #1841 for details.

This PR brings the `IsInRangeConverter` back into alignment with the `CompareConverter`, in terms of implementation.

The current use of the converter is as follows:

```xaml
            <Color x:Key="LightGreen">LightGreen</Color>
            <Color x:Key="PaleVioletRed">PaleVioletRed</Color>
            <x:Double x:Key="MinValue">3.0</x:Double>
            <x:Double x:Key="MaxValue">7.0</x:Double>
            <converters:DoubleIsInRangeConverter
                x:Key="isBetweenThreeAndSevenConverter"
                FalseObject="{StaticResource PaleVioletRed}"
                MaxValue="{StaticResource MaxValue}"
                MinValue="{StaticResource MinValue}"
                TrueObject="{StaticResource LightGreen}" />
```
If a developer tries to write `MaxValue="7"`, and the `value` for the converter is a `Double`, the user will not get any valid results from the converter.  In XAML, the `MaxValue="7"` has a type of `String`, by default.

This PR makes it possible to define what type `MaxValue` and `MinValue` should be when inheriting from the converter, based on the `value`.

### Converter definition ###
```csharp
public sealed class DoubleIsInRangeConverter : IsInRangeConverter<double, Color>;
```
### XAML Usage ###
```xaml
            <converters:DoubleIsInRangeConverter
                x:Key="isBetweenThreeAndSevenConverter"
                FalseObject="{StaticResource PaleVioletRed}"
                MaxValue="7"
                MinValue="3"
                TrueObject="{StaticResource LightGreen}" />
```

I have also removed the un-used `CompareValue` property.

I think that this will make is easier for users to use, and eliminate any confusion about XAML defaulting to `String` for this control.

 ### Linked Issues ###
 

 - Fixes #

 ### PR Checklist ###
 
 - [ ] Has a linked Issue, and the Issue has been `approved`(bug) or `Championed` (feature/proposal)
 - [X] Has tests (if omitted, state reason in description)
 - [X] Has samples (if omitted, state reason in description)
 - [X] Rebased on top of `main` at time of PR
 - [X] Changes adhere to [coding standard](https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#contributing-code---best-practices)
 - [ ] Documentation created or updated: https://github.com/MicrosoftDocs/CommunityToolkit/pulls 

 ### Additional information ###

### CODE REVIEW

**Comment by GeorgeLeithead** [comment]
Note that I would need to update the DOCs, so this should be blocked until the docs are ready.

**Comment by GeorgeLeithead** [comment]
This also "Needs Discussion" label added.

File: src/CommunityToolkit.Maui/Converters/IsInRangeConverter.shared.cs
Comment (bijington) [inline src/CommunityToolkit.Maui/Converters/IsInRangeConverter.shared.cs:16]: This is a breaking change, are we sure we want to drop the ability to make the following properties bindable?

- `FalseObject`
- `MaxValue`
- `MinValue`
- `TrueObject`

File: src/CommunityToolkit.Maui/Converters/IsInRangeConverter.shared.cs
Comment (bijington) [inline src/CommunityToolkit.Maui/Converters/IsInRangeConverter.shared.cs:15]:
```suggestion
	/// <summary>If supplied this value will be returned when the converter receives an input value that is <b>outside</b> of the <see cref="MinValue" /> and <see cref="MaxValue" />s.</summary>
```

File: src/CommunityToolkit.Maui/Converters/IsInRangeConverter.shared.cs
Comment (bijington) [inline src/CommunityToolkit.Maui/Converters/IsInRangeConverter.shared.cs:24]:
```suggestion
	/// <summary>If supplied this value will be returned when the converter receives an input value that is <b>inside</b> (inclusive) of the <see cref="MinValue" /> and <see cref="MaxValue" />s.</summary>
```

File: src/CommunityToolkit.Maui/Converters/IsInRangeConverter.shared.cs
Comment (bijington) [inline src/CommunityToolkit.Maui/Converters/IsInRangeConverter.shared.cs:18]:
```suggestion
	/// <summary>The upper bounds of the range to compare against when determining whether the input value to the convert is within range.</summary>
```

File: src/CommunityToolkit.Maui/Converters/IsInRangeConverter.shared.cs
Comment (bijington) [inline src/CommunityToolkit.Maui/Converters/IsInRangeConverter.shared.cs:21]:
```suggestion
	/// <summary>The lower bounds of the range to compare against when determining whether the input value to the convert is within range.</summary>
```

**Review by bijington — CHANGES_REQUESTED** [review]
@GeorgeLeithead thank you for this! Apologies for the delay in reviewing this. I have added a number of comments, some improving existing bits (xml docs) and others just to confirm my understanding. I like the concept.

File: src/CommunityToolkit.Maui/Converters/IsInRangeConverter.shared.cs
Comment (GeorgeLeithead) [reply src/CommunityToolkit.Maui/Converters/IsInRangeConverter.shared.cs:46]: Have removed the unnecessary check.

File: src/CommunityToolkit.Maui/Converters/IsInRangeConverter.shared.cs
Comment (GeorgeLeithead) [reply src/CommunityToolkit.Maui/Converters/IsInRangeConverter.shared.cs:51]: Have removed the unnecessary check.

File: src/CommunityToolkit.Maui/Converters/IsInRangeConverter.shared.cs
Comment (GeorgeLeithead) [reply src/CommunityToolkit.Maui/Converters/IsInRangeConverter.shared.cs:71]: Have removed the unnecessary method.

File: src/CommunityToolkit.Maui/Converters/IsInRangeConverter.shared.cs
Comment (GeorgeLeithead) [reply src/CommunityToolkit.Maui/Converters/IsInRangeConverter.shared.cs:1]: I have removed the 'Ignore Spelling'.

File: src/CommunityToolkit.Maui/Converters/IsInRangeConverter.shared.cs
Comment (GeorgeLeithead) [reply src/CommunityToolkit.Maui/Converters/IsInRangeConverter.shared.cs:16]: I was copying the changes that you made to `CompareConverter` to bring this converter in-to-line.  As such, I believe it will have the same impact.

File: src/CommunityToolkit.Maui.UnitTests/Converters/IsInRangeConverterTests.cs
Comment (VladislavAntonyuk) [inline src/CommunityToolkit.Maui.UnitTests/Converters/IsInRangeConverterTests.cs:11]: should be Local?

Comment (GeorgeLeithead) [reply src/CommunityToolkit.Maui.UnitTests/Converters/IsInRangeConverterTests.cs]: You are right.  I have re-named the variable to `UtcDateTimeKind`.

**Review by bijington — DISMISSED** [review]
Thank you for this @GeorgeLeithead

**Comment by brminnick** [comment]
Are we sure that removing the all of the `BindableProperty`s is a good choice? 

This means that users can no longer bind to any of the following properties from the VIewModel:
* FalseObject
* MaxValue
* MinValue
* TrueObject

Personally, I'd prefer that we keep the bindable properties.

**Comment by bijington** [comment]
> Are we sure that removing the all of the `BindableProperty`s is a good choice? 
> 
> 
> 
> This means that users can no longer bind to any of the following properties from the VIewModel:
> 
> * FalseObject
> 
> * MaxValue
> 
> * MinValue
> 
> * TrueObject
> 
> 
> 
> Personally, I'd prefer that we keep the bindable properties unless someone with more XAML knowledge can help me understand.

That was my initial thinking but I struggled to get input on this.

**Comment by bijington** [comment]
Let me put them back so we can merge this in

**Review by brminnick — APPROVED** [review]

**Comment by GeorgeLeithead** [comment]
I have some comments to make on this. Well do so tomorrow.

**Comment by GeorgeLeithead** [comment]
Um.. Ok, too late to add my 2-pennies... but...

- The original implementation I made for `IsInRangeConverter` was based on the ONLY other `generic` converter in the toolkit `CompareConverter`.  When it was upgraded in PR #1841 , this therefore prompted me to 'do the same' to this converter.  As such, the bindable properties were not present in the `CompareConverter`.
- All of the samples, tests, examples in the docs AND the examples I provided in the PR details, all work and in fact it's now much easier now as it's easier to provide the TYPE.
- Currently, I believe that no other converter in the tool kit provides any `BindableProperty`(s).

So, I think the big question in reality is:  Should we either adhere to the same standard and NOT have bindable properties, or should we add bindable properties to **all** [appropriate] converters?  If the former, then I don't think we should have bindable properties on this converter.  If the latter, then IMO a bigger PR should be undertaken, and at the (very IMO) VERY LEAST additional Unit Tests should have been added BEFORE we approved the PR.

Thoughts? @bijington @brminnick, et all? (Should this be an open discussion on Discord, instead of against the PR?)

**Pull request merged**
